### PR TITLE
Extensibility: Details view: Hide key if the value is empty from panels

### LINKED ISSUE #1539: Extensibility: Details view: Hide key if the value is empty from panels
**Description**
I want to be able to hide the full entries from panels or other components when the value is empty. So instead of showing a key with an empty value to the right side, it would not show the key at all. I've attached 2 screenshots below to explain what I mean:

**Current**
image

**Wanted**
image

**Reasons**
We want to hide fields which are empty and not relevant to the user when viewing our resources. There is also the case that we have fields which are mutually exclusive so that only one of the two will ever be there and we want to only display the relevant one.

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- As in issue.
- Add more tests

**Related issue(s)**

Closes #1539

### CODE REVIEW

File: docs/extensibility/resources.md [hunk 1]
```diff
@@ -106,6 +106,10 @@ The `details` section defines the display structure for the details page. It con
 - **widget** - optional widget to render the defined entry. By default the value is displayed verbatim. For more information about the available widgets, see [Display widgets](display-widgets.md).
 - **valuePreprocessor** - name of [value preprocessor](#value-preprocessors),
 - **formula** - optional formula used to modify data referred to by the `path` property. To learn more about using formulas, see [JSONata](https://docs.jsonata.org/overview.html).
+- **visibility** - by default all fields are visible; however `visibility` property can be used to control a single item display.
+  - If set to `false` explicitly, the field won't render.
```
Comment (grego952) [inline docs/extensibility/resources.md:110]:
```suggestion
  - If set to `false` explicitly, the field doesn't render.
```

File: docs/extensibility/resources.md [hunk 2]
```diff
@@ -106,6 +106,10 @@ The `details` section defines the display structure for the details page. It con
...
+  - If set to any string, this property will be treated as jsonata formula, determining (based on current value given as `data`) if the field should be visible.
```
Comment (grego952) [inline docs/extensibility/resources.md:111]:
```suggestion
  - If set to any string, this property is treated as jsonata formula, determining (based on current value given as `data`) if the field should be visible.
```

File: docs/extensibility/resources.md [hunk 3]
```diff
@@ -106,6 +106,10 @@ The `details` section defines the display structure for the details page. It con
...
+  - If not set the field will always be rendered.
```
Comment (grego952) [inline docs/extensibility/resources.md:112]:
```suggestion
  - If not set, the field always renders.
```

File: docs/extensibility/resources.md [hunk 4]
```diff
@@ -106,6 +106,10 @@ The `details` section defines the display structure for the details page. It con
 - **widget** - optional widget to render the defined entry. By default the value is displayed verbatim. For more information about the available widgets, see [Display widgets](display-widgets.md).
 - **valuePreprocessor** - name of [value preprocessor](#value-preprocessors),
 - **formula** - optional formula used to modify data referred to by the `path` property. To learn more about using formulas, see [JSONata](https://docs.jsonata.org/overview.html).
+- **visibility** - by default all fields are visible; however `visibility` property can be used to control a single item display.
```
Comment (grego952) [inline docs/extensibility/resources.md:109]:
```suggestion
- **visibility** - by default all fields are visible; however **visibility** property can be used to control a single item display.
```
if visibility is a parameter it should be bolded

File: docs/extensibility/resources.md [hunk 5]
```diff
@@ -106,6 +106,10 @@ The `details` section defines the display structure for the details page. It con
 - **widget** - optional widget to render the defined entry. By default the value is displayed verbatim. For more information about the available widgets, see [Display widgets](display-widgets.md).
 - **valuePreprocessor** - name of [value preprocessor](#value-preprocessors),
 - **formula** - optional formula used to modify data referred to by the `path` property. To learn more about using formulas, see [JSONata](https://docs.jsonata.org/overview.html).
+- **visibility** - by default all fields are visible; however **visibility** property can be used to control a single item display.
```
Comment (wchrapka) [inline docs/extensibility/resources.md:109]: JSONata tested successfully for Panel children, doesn't work for other components:
CodeViewer - for `"path": "spec.files",` requires `"visibility": "$exists(data.spec.files)"`
Text - for `"path": "kind"` requires `"visibility": "$exists(data.kind)"`
same for the widgets in the header
etc.

File: core-ui/src/setupTests.js [hunk 6]
```diff
@@ -53,6 +53,24 @@ jest.mock('graphviz-react', () => ({
   Graphviz: () => 'Graphviz mock',
 }));
```
Comment (wchrapka) [inline core-ui/src/setupTests.js:54]: @dariadomagala is working on the tests, maybe she wants a say.

File: core-ui/src/shared/components/ResourceDetails/ResourceDetails.js [hunk 7]
```diff
@@ -290,6 +290,17 @@ function Resource({
     }
   };
 
+  const filterColumns = col => {
+    const { visible, error } = col.visibility?.(resource) || {
```
Comment (wchrapka) [inline core-ui/src/shared/components/ResourceDetails/ResourceDetails.js:294]: didn't we agree we don't need it for columns?

**Review by wchrapka — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**